### ISSUE TITLE
Opened channel, spent some dust amounts of satoshi's, closed channel, got spend satoshi's back

### ISSUE THREAD

**Issue by kristapsk** [lightningnetwork/lnd#917]
TL;DR - (Bitcoin mainnet) opened channel, did some small few hundred transactions, later closed channel, got back all my money minus channel open and close TX fees (so, total walletbalance was more than I had as walletbalance + channel's local_balance before closing - closing TX fee).

What I did.

March 20:

1) Installed LND 0.4 beta (with ce856323906e4c7eafd55de8500e75f42fa93ca0 patch applied, I route all DNS requests via Tor on that particual box, so only A/AAAA records are supported), funded wallet with amount A;
2) Opened channel with yalls.org, using some amount B (so we have A - amount B - change amount C - opening TX fees)
3) Did small (150 sat) payment directly to yalls.org as a test
4) Tried to buy stuff from Blockstream Store, [unsuccessfully](https://github.com/lightningnetwork/lnd/issues/643#issuecomment-374441036)
5) Found some random site to which I paid 100 satoshis to play some short game with 1 satoshi routing fee (to test it's really problem at the Blockstream side, not at mine or yalls.org)
6) Shut down my LND instance

March 21 (about 24 hours later):

7) Started my LND node again
8) Tried checkout at Blockstream Store again, this time got ["unable to route payment to destination: TemporaryChannelFailure: insufficient capacity in available outgoing links"](https://github.com/lightningnetwork/lnd/issues/643#issuecomment-375094999) error
9) Decided it's good time to test `lncli closeallchannels`
10) Looked at channel closing TX which had all the channel's local_balace + remote_balance as the input and only one (mine) TX output
11) Transaction confirms, I have amount B + amount C - close TX fee in my walletbalance, I succesfully stole 251 satoshi's from two other parties...

I can provide involved parties and LND dev's (@Roasbeef?) with TX id's and other necessary information.

**Comment by Roasbeef** [comment]
Can you replace the title with something more descriptive?

**Comment by Roasbeef** [comment]
If you have dust outputs, or the remote party did, then those aren't manifested on the commitment transaction.

**Comment by Roasbeef** [comment]
The closing tx only uses the multi-sig outputs as inputs. If the remote party didn't have a balance above your dust limit, then it's sent to miner's fee in order to ensure wide propagation of the closing transaction which respects the common mempool policy of node implementations.

**Comment by kristapsk** [comment]
Ok, I had that idea in my mind at one point, before filling this issue. So it means, dust amount payments can't be reliable on LN? Probably I should not be allowed to close such channel so easely unilatery? (haven't studied all the LN specs yet, sorry)

**Comment by Roasbeef** [comment]
Both parties set a dust limit on their commitments. Both parties can have different dust limits. If an output is below the dust limit, then we don't add it to the commitment transaction, and instead it goes to miner's fees. When making channels, either side can set higher or lower dust limits as they wish. This is outside the scope of LN, as dust exists as a miner mempool policy. It's possible, but very difficult in practice (unless you send to miners directly) to create and then spend a dust output, so we avoid it entirely.

**Comment by Roasbeef** [comment]
> dust amount payments can't be reliable on LN

Dust payments aren't reliable on _Bitcoin_. Within a channel, you can accept them, then possibly even re-balance your channels in order to close out the amount, and add the dust value to your balance.

**Comment by Roasbeef** [comment]
If you check your balances, the spent satoshis _wouldn't_ have been restored. Just that they were sent to miner's fees.

**Comment by Roasbeef** [comment]
That second payment didn't go through as you didn't have enough capacity.

**Comment by kristapsk** [comment]
> That second payment didn't go through as you didn't have enough capacity.

Which second payment? To the Blockstream Store? No, there I had more than enough capactity (I especially funded my channel with some extra ~10 EUR worth of BTC more I wanted to spend), that amount listed below needed there wasn't in my channel.

**Comment by Roasbeef** [comment]
Your error was: 
> TemporaryChannelFailure: insufficient capacity in available outgoing links: need 326352390 mSAT, max available is 67156600 mSAT

It could've been the case that that direct channel with the Blockstream Store wasn't active, so instead `lnd` tried to route via another channel, which didn't have enough capacity.

**Comment by kristapsk** [comment]
> another channel, which didn't have enough capacity.

I understood that. So decided, as I am anyway testing this thing, let's try to close channel and then open some different channel to different party later and test again. Then, after successful closing of channel, I looked at all the numbers, they didn't round up for me and opened this issue.

But, ok, you explained more or less, seems to be no issue, at least for now.

**Comment by kristapsk** [comment]
Searching for Blockstream node id and opening channel to them directly would be too boring. :)

**Comment by Roasbeef** [comment]
Well as I've noted on other issues, their store is running an old version of `clightning`, and their latest version has had a ton of bugs fixed and has general improvements. So you may have a hard time reaching them in general until they update.

**Comment by kristapsk** [comment]
At least the reaction was different than 24 hours ago. :) Actually, I first tried Lightning with c-lightning few weeks ago (mainly because LND didn't support bitcoind backend back then), but wasn't able to get it running back then. Probably I should try that with new master version again.
